src/DataStructures/ModalVector.hpp

```diff
@@ -0,0 +1,109 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for storing spectral coefficients on a spectral grid.
+ *
+ * \details A ModalVector holds an array of spectral coefficients, and can be
+ * either owning (the array is deleted when the ModalVector goes out of scope)
+ * or non-owning, meaning it just has a pointer to an array.
+ *
+ * Only basic mathematical operations are supported with
+ * ModalVectors. ModalVectors may be added or subtracted, and the following
+ * unary operations are supported:
+ * - abs
+ * - max
+ * - min
+ *
+ * Also, addition, subtraction, multiplication, and division are supported with
+ * doubles and `ModalVector`s and multiplication is supported with
```

Comment:
Does double + ModalVector make sense?

Reply:
No, it can't, same with subtraction.

Reply:
Also it should be clarified that only ModalVector/double is supported